components/speedreader/rust/lib/src/readability/src/extractor.rs

```diff
@@ -327,6 +327,15 @@ pub fn post_process(dom: &mut Sink, root: Handle, meta: &Meta) {
             }
         }
 
+        // Add 'show original'
+        {
+            let show_original_link =
+                dom::create_element_simple(dom, "a", "subhead metadata", None);
+            dom::set_attr("id", "show-original-page", show_original_link.clone(), true);
```

Comment:
consider calling the ID something more unique in case the page already has an element with ID `show-original-page`